### PR TITLE
chore: rename `holochain_deterministic_integrity` to `hdi`

### PR DESCRIPTION
**Pull request by steveeJ**
### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,33 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 # \[Unreleased\]
 
+* **BREAKING**: the `hdi` crate has been renamed to `hdi`
```
Comment (Connoropolous) [inline CHANGELOG.md:9]: Isn't it the holochain_deterministic_integrity crate which has been renamed?

Comment (steveeJ) [reply CHANGELOG.md:9]: thank you for catching this. i ran the rename command again after writing the change log so it was subject to the rename :laughing:

File: crates/hdi/Cargo.toml [hunk 2]
```diff
@@ -25,7 +25,7 @@ hdk_derive = { version = "0.0.40", path = "../hdk_derive" }
 holo_hash = { version = "0.0.30", path = "../holo_hash" }
 holochain_wasmer_guest = "=0.0.80"
 # it's important that we depend on holochain_integrity_types with no default
-# features, both here AND in holochain_deterministic_integrity_derive, to reduce code bloat
+# features, both here AND in hdk_derive, to reduce code bloat
```
Comment (steveeJ) [inline crates/hdi/Cargo.toml:28]: i'm not sure whether this is accurate but the original _holochain_deterministic_integrity_derive_ that was in place here is a non-existing crate

Comment (jost-s) [reply crates/hdi/Cargo.toml:28]: the only `_derive` crate there is

@freesig Is that the correct replacement?

**Review by jost-s — APPROVED** [review]
LGTM

**Pull request merged**